ocrd_network/ocrd_network/processing_worker.py

```diff
@@ -0,0 +1,327 @@
+        # Verify that the processor name in the processing message
+        # matches the processor name of the current processing worker
+        if self.processor_name != processing_message.processor_name:
+            raise ValueError(f'Processor name is not matching. Expected: {self.processor_name},'
+                             f'Got: {processing_message.processor_name}')
+
+        # This can be path if invoking `run_processor`
+        # but must be ocrd.Workspace if invoking `run_cli`.
+        path_to_mets = processing_message.path_to_mets
+        # Build the workspace from the workspace_path
+        workspace = Resolver().workspace_from_url(path_to_mets)
+
+        job_id = processing_message.job_id
+        sync_set_processing_job_state(job_id=job_id, job_state=StateEnum.running)
+        if self.processor_class:
+            self.log.debug(f'Invoking the pythonic processor: {self.processor_name}')
+            return_status = self.run_processor_from_worker(
+                processor_class=self.processor_class,
+                workspace=workspace,
+                page_id=processing_message.page_id,
+                input_file_grps=processing_message.input_file_grps,
+                output_file_grps=processing_message.output_file_grps,
+                parameter=processing_message.parameters
+            )
+        else:
+            self.log.debug(f'Invoking the cli: {self.processor_name}')
+            return_status = self.run_cli_from_worker(
+                executable=self.processor_name,
+                workspace=workspace,
+                page_id=processing_message.page_id,
+                input_file_grps=processing_message.input_file_grps,
+                output_file_grps=processing_message.output_file_grps,
+                parameter=processing_message.parameters
+            )
+        job_state = StateEnum.success if return_status else StateEnum.failed
+        sync_set_processing_job_state(job_id=job_id, job_state=job_state)
+
+        result_queue_name = processing_message.result_queue_name
+        callback_url = processing_message.callback_url
+
+        if result_queue_name or callback_url:
+            result_message = OcrdResultMessage(
+                job_id=str(job_id),
+                status=job_state.value,
+                # Either path_to_mets or workspace_id must be set (mutually exclusive)
+                path_to_mets=path_to_mets,
+                workspace_id=None
+            )
+            self.log.info(f'Result message: {result_message}')
+
+        # If the result_queue field is set, send the result message to a result queue
+        if result_queue_name:
+            self.publish_to_result_queue(result_queue_name, result_message)
+
+        # If the callback_url field is set, post the result message to a callback url
+        if callback_url:
+            self.post_to_callback_url(processing_message.callback_url, result_message)
```

Comment:
I would recommend to indent these code one more level, so put them under the `if result_queue_name or callback_url:` check above.